### PR TITLE
Migrate e2e test for Cassandra to Go

### LINKED ISSUE #3231: Migrate e2e test for Cassandra to Go
### Proposal

Relates to https://github.com/kedacore/keda/issues/2357, https://github.com/kedacore/keda/issues/2737.

### Use-Case

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by aviadlevy**
Signed-off-by: aviadlevy <aviadlevy1@gmail.com>

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added

Fixes #3231

### CODE REVIEW

**Comment by aviadlevy** [comment]
So glad it passed. I only forgot to delete the old test

**Comment by aviadlevy** [comment]
> /run-e2e cassandra* **Update:** You can check the progress [here](https://github.com/kedacore/keda/actions/runs/2594742378)

Seems like something was wrong with initiating the scaled object - https://github.com/kedacore/keda/runs/7144994425?check_suite_focus=true#step:6:220 

Can you re-run the e2e?

P.S.

I noticed that the test cleanup is finished before all the resources are actually cleaned - https://github.com/kedacore/keda/runs/7144994425?check_suite_focus=true#step:6:541 
With this behavior there is no point to retry. I opened an issue to fix that - https://github.com/kedacore/keda/issues/3322

File: tests/cleanup_test.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ func TestRemoveWorkloadIdentityComponents(t *testing.T) {
 	_, err := ExecuteCommand(fmt.Sprintf("helm uninstall workload-identity-webhook --namespace %s", AzureWorkloadIdentityNamespace))
 	require.NoErrorf(t, err, "cannot uninstall workload identity webhook - %s", err)
 
-	Kc = GetKubernetesClient(t)
+	Kc := GetKubernetesClient(t)
```
Comment (JorTurFer) [inline tests/cleanup_test.go:31]: Maybe we should use `KubeClient` instead of `Kc` to unify the naming

File: tests/setup_test.go [hunk 2]
```diff
@@ -68,7 +68,7 @@ func TestSetupWorkloadIdentityComponents(t *testing.T) {
 	_, err = ExecuteCommand("helm repo update azure-workload-identity")
 	require.NoErrorf(t, err, "cannot update workload identity helm repo - %s", err)
 
-	Kc = GetKubernetesClient(t)
+	Kc := GetKubernetesClient(t)
```
Comment (JorTurFer) [inline tests/setup_test.go:71]: same as above

**Review by JorTurFer — COMMENTED** [review]
Looking good, small nits inline

**Review by JorTurFer — APPROVED** [review]
LGTM!
Thanks ❤️

**Pull request merged**